tests/cpp/test_allocation_order_inference.cpp

```diff
@@ -133,13 +133,32 @@ TEST_F(AllocationOrderInferenceTest, BinaryOpPropagation) {
     tv1->setAllocationDomain(tv1_format, true);
 
     preseg_passes::inferenceAllocationOrder(&fusion, {tv0, tv1}, {tv2, tv3});
-    // tv1 dominates output allocation order, which has a permutation {1, 0, 2,
-    // 3}. But since tv1->axis(3) is a broadcast dimension, it did not map to
-    // tv2->axis(3)/tv3->axis(3). Propagated permutation would push the unmapped
-    // axis(3) first in the allocation domain while keeping mapped ids in its
-    // original order {1, 0, 2} as inner entries in its allocation domain.
-    EXPECT_THAT(getAllocationDomainPermutation(tv2), ElementsAre(3, 1, 0, 2));
-    EXPECT_THAT(getAllocationDomainPermutation(tv3), ElementsAre(3, 1, 0, 2));
+    EXPECT_THAT(getAllocationDomainPermutation(tv2), ElementsAre(1, 0, 2, 3));
+    EXPECT_THAT(getAllocationDomainPermutation(tv3), ElementsAre(1, 0, 2, 3));
+  }
+  {
+    auto fusion_ptr = std::make_unique<Fusion>();
+    Fusion& fusion = *fusion_ptr.get();
+    FusionGuard fg(&fusion);
+
+    // Testing propagation between two tensors
+    // tv0 has more non-broadcast iter domain and dominates output memory format
+    auto tv0 = makeSymbolicTensor({1, -1, -1, -1});
+    fusion.addInput(tv0);
+    auto tv1 = makeSymbolicTensor({-1, 1, 1, 1});
+    fusion.addInput(tv1);
+    auto tv2 = add(tv0, tv1);
+    fusion.addOutput(tv2);
+
+    // since tv0->axis(0) is a broadcast, and tv2->axis(0) is not exact map.
+    // The mapping of inner dimensions stops here.
+    // tv2 will have output allocation order as {0, 2, 3, 1}.
```

Comment:
Sorry, I always get confused with these examples. I don't doubt it's correct but I want to understand this. `tv0` is broadcast in first dim and iteration in all inner dims, so it exact maps to `tv2` in the three innermost logical dims but not the outer logical dim. In the allocation domain of `tv0`, we have one innermost exact-mapped domain, and as expected that domain is innermost in `tv2`s propagated allocation domain. Since the propagation stops at the next inner allocation axis `tv0->axis(0)`, the remaining axes of `tv2`'s allocation domain are inserted in their original order. 

Why is this preferable to moving `tv0->axis(0)`, which is a broadcast axis and can be placed anywhere, to the outermost position of `tv0` when doing the propagation, so that `tv2`'s order would be `0, 3, 2, 1`?

Reply:
> Why is this preferable to moving tv0->axis(0), which is a broadcast axis and can be placed anywhere, to the outermost position of tv0 when doing the propagation, so that tv2's order would be 0, 3, 2, 1?

I'm just being conservative on reordering and that might be a bit too much for this example. I do not have a good concrete example against that right now.
I think the suggested order would allow us to higher vectorization factor, which is preferable.

We can try your suggestion. Which should be straighforward by adding a `IdMappingMode::PERMISSIVE` in the loop structure. :bow: